### ISSUE TITLE
Calls to println/eprintln

### ISSUE THREAD

**Issue by ettom** [njaard/sonnerie#32]
It would be nice if the library didn't just print stuff to stdout/stderr indiscriminately. AFAIK the idiomatic solution would be to use the `log` or `tracing` crate.

**Comment by njaard** [comment]
I agree! There are only a few cases of `eprintln` in the code. Many of them can be removed harmlessly.

The "disregarding database" error should be part of the user interface.

The biggest question is what to do about `warning: invalid segment version`. Is that a corrupt or malformed file a valid use for a panic?

**Comment by ettom** [comment]
I think that the library itself should *never* call `(e)println` and ideally it wouldn't panic either. In both cases, doing otherwise is taking control away from the application developer. What about propagating the `disregarding database` or `invalid segment version` and similar errors as `Err`s, with a custom Error type? For this, the `thiserror` create could help. You could have different variants with some of them being marked as recoverable and others as fatal, where appropriate.
